File: backend/models/VideoStream.js
const VideoStream = {
  title: {
    type: String,
    trim: true,
    default: "unknown",
  },

  index: {
    type: Number,
    required: [true, "index is required"],
  },

  profile: {
    type: String,
    required: [true, "profile is required"],
  },

  bitDepth: {
    type: String,
    required: [true, "bit depth is required"],
  },

  level: {
    type: Number,
    required: [true, "level is required"],
  },

  codec: {
    type: String,
    required: [true, "codec is required"],
  },

  width: {
    type: Number,
    required: [true, "width is required"],
  },

  height: {
    type: Number,
    required: [true, "height is required"],
  },

  codedWidth: {
    type: String,
    required: [true, "coded width is required"],
  },

  codedHeight: {
    type: String,
    required: [true, "coded height is required"],
  },

  aspectRatio: {
    type: String,
    default: "unknown",
    trim: true,
  },

  colorTransfer: {
    type: String,
    default: "unknown",
    trim: true,
  },

  colorSpace: {
    type: String,
    default: "unknown",
    trim: true,
  },

  colorPrimaries: {
    type: String,
    default: "unknown",
    trim: true,
  },

  frameRate: {
    type: String,
    default: 0,
  },

  avgFrameRate: {
    type: String,
    default: 0,
  },

  bps: {
    type: Number,
    default: 0,
  },

  numberOfFrames: {
    type: Number,
    default: 0,
  },

  numberOfBytes: {
    type: Number,
    default: 0,
  },
};

export default VideoStream;

File: backend/cmd/models/VideoStream.go
package models

import "gorm.io/gorm"

type VideoStream struct {
	gorm.Model
	Title          string `gorm:"not null;default:'unknown'" json:"title"`
	Index          uint   `json:"index"`
	Profile        string `gorm:"not null;default'unknown'" json:"profile"`
	AspectRatio    string `gorm:"not null;default:'unknown'" json:"aspectRatio"`
	BitDepth       string `gorm:"not null;default:'unknown'" json:"bitDepth"`
	Codec          string `gorm:"not null" json:"codec"`
	Width          uint   `json:"width"`
	Height         uint   `json:"height"`
	CodedWidth     uint   `json:"codedWidth"`
	CodedHeight    uint   `json:"codedHeight"`
	ColorTransfer  string `gorm:"not null;default:'unknown'" json:"colorTransfer"`
	ColorPrimaries string `gorm:"not null;default:unknown'" json:"colorPrimaries"`
	ColorSpace     string `gorm:"not null;default:'unknown'" json:"colorSpace"`
	FrameRate      string `gorm:"not null;default:'unknown'" json:"frameRate"`
	AvgFrameRate   string `gorm:"not nulldefault:'unknown'" json:"avgFrameRate"`
	BPS            uint   `json:"bps"`
	MovieID        uint   `json:"movieID"`
	Movie          Movie  `json:"movie"`
}
